### PR TITLE
Df 1136 jobs api filtering for status

### PR DESCRIPTION
**Pull request by dvoracek-martin**

### CODE REVIEW

File: web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/api/JobAdvertisementApiRestController.java [hunk 1]
```diff
@@ -80,8 +82,10 @@ public JobAdvertisementApiRestController(JobAdvertisementApplicationService jobA
      * - 403 Forbidden: User has not the required permission to perform this action
      */
     @PostMapping("/_search")
-    public PageResource<ApiJobAdvertisementDto> findJobAdvertisementsByStatus(@RequestBody @Valid ApiSearchRequestDto apiSearchRequestDto, @RequestParam(name = "page", defaultValue = "0") int page, @RequestParam(name = "size", defaultValue = "25") int size) {
-        final PageRequest pageRequest = PageRequest.of(page, size, Sort.by(Sort.Order.desc("createdTime")));
+    public PageResource<ApiJobAdvertisementDto> findJobAdvertisementsByStatus(@RequestBody @Valid ApiSearchRequestDto apiSearchRequestDto, @RequestParam(name = "page", defaultValue = "0") int page, @RequestParam(name = "size", defaultValue = "25") int size, @RequestParam(name = "sort", defaultValue = "desc") String sort) {
```
Comment (pavax) [inline web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/api/JobAdvertisementApiRestController.java]: 1. create a enum for the sorting and use that one as a parameter for spring-mvc
2. did you check that the updatedTime is never null

File: application/src/test/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/JobAdvertisementApplicationServiceForAPITest.java [hunk 2]
```diff
@@ -113,16 +113,19 @@ public void searchFromApi() {
...
+        apiSearchRequestDto.setStatus(statuses);
+        final PageRequest pageRequest = PageRequest.of(0, 25, Sort.by(Sort.Order.desc("updatedTime")));
+      
         //when
         Page<JobAdvertisementDto> jobAdvertisementDtos = service.findJobAdvertisementsByStatus(pageRequest, apiSearchRequestDto);
 
         //then
         assertThat(jobAdvertisementDtos.getContent()).isNotNull();
         assertThat(jobAdvertisementDtos.getContent()).isNotEmpty();
+        assertThat(jobAdvertisementDtos.getContent().size() == 1);
```
Comment (pavax) [inline application/src/test/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/JobAdvertisementApplicationServiceForAPITest.java]: use assertThat(xxxx).isEqualTo(1)

File: domain/src/test/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/fixture/OwnerFixture.java [hunk 3]
```diff
@@ -10,7 +10,7 @@
     public static Builder of(JobAdvertisementId id){
         return testOwner()
                 .setAccessToken(format("access-token-%s", id.getValue()))
-                .setUserId(format("user-id-%s", id.getValue()))
+                .setUserId(format("api-user-id"))
```
Comment (pavax) [inline domain/src/test/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/fixture/OwnerFixture.java]: why did you change that

File: web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/api/Sort.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package ch.admin.seco.jobs.services.jobadservice.infrastructure.web.controller.api;
+
+public enum Sort {
+    DATE_ASC ("asc"),
```
Comment (pavax) [inline web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/api/Sort.java]: you can get rid of the string value once you add the Sort as a request-parameter. spring-mvc is going to take care of the rest

File: domain/src/test/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/fixture/OwnerFixture.java (hunk 3 above)
Comment (dvoracek-martin) [reply domain/src/test/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/fixture/OwnerFixture.java]: To have a fixed value. I am running the test as the api-user and the search query has a condition that the job ad has to be owned bu current user or company

File: application/src/test/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/JobAdvertisementApplicationServiceForAPITest.java (hunk 2 above)
Comment (dvoracek-martin) [reply application/src/test/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/JobAdvertisementApplicationServiceForAPITest.java]: done

File: web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/api/JobAdvertisementApiRestController.java (hunk 1 above)
Comment (dvoracek-martin) [reply web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/api/JobAdvertisementApiRestController.java]: 1) done
2) Yes. Update time == create time , when a new job ad is created
3) done

File: web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/api/Sort.java (hunk 4 above)
Comment (dvoracek-martin) [reply web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/api/Sort.java]: done
